sqlmesh/core/loader.py

```diff
@@ -272,13 +264,21 @@ def _load_models(
         models.update(self._load_external_models())
         models.update(self._load_python_models())
 
+        # Make sure all models are unique by their name in addition to their FQN
+        seen_names: t.Set[str] = set()
+        for model in models.values():
+            if model.name in seen_names:
+                raise ConfigError(
+                    f"Model name '{model.name}' is not unique. Please make sure all models have unique names."
```

Comment:
how can this ever happen? why would you have a unique fqn but a non unique model name?

can you have a default catalog at the model level?

Reply:
You technically can have default catalog at the model level and that is how this can happen. I personally don't like that but it is how you originally implemented. If you open to changing then we can discuss and this can be removed.

Reply:
ok, let's chat about this

Reply:
i think we've decided to get rid of this because model default catalog won't be overridable

Reply:
Yep removed the override so this is now removed.